scriptmodules/emulators/jzintv.sh

```diff
@@ -61,17 +70,60 @@ function install_jzintv() {
 }
 
 function configure_jzintv() {
-    mkRomDir "intellivision"
+    local -r system="intellivision"
+    mkRomDir $system
 
-    local options=(
-        --displaysize="%XRES%x%YRES%"
-        --quiet
-        --rom-path="$biosdir"
-        --voice=1
-    )
+    local start_script="$md_inst/jzintv_launcher.sh"
+    cat > "$start_script" << _EOF_
+#! /usr/bin/env bash
+
+# \$1: width of display
+# \$2: height of display
+# \$3: --ecs=1, optional
+# \$4,5,6...: more optional parameters
+# last parameter: %ROM%
+
+jzintv_bin="$md_inst/bin/jzintv"
+
+disp_w=\$1; shift
+disp_h=\$1; shift
+
+min=\$(( \$disp_w < \$disp_h ? \$disp_w : \$disp_h ))
+if [[ \$min = \$disp_h ]] ; then
+    intv_w=\$(echo \$min | awk '{ printf "%0.f\n", \$0 * 1.3333 + .5 }')
+    intv_h=\$min
+else
+    # 90/270 rotated displays
+    intv_w=\$min
+    intv_h=\$(echo \$min | awk '{ printf "%0.f\n", \$0 / 1.3333 }')
+fi
+
+# set --gfx-verbose instead of --quiet for verbose output
+options=(
+    -f1 # fullscreen
+    --quiet
+#    --gfx-verbose
+    --displaysize="\${intv_w}x\${intv_h}"
+    --rom-path="$biosdir"
+    --voice=1
+)
+
+# ingest additional options
+while [[ \$# -gt 1 ]] ; do
+    options+=(\$1); shift
+done
+
+rom=\$1
+echo "Launching: \$jzintv_bin \${options[*]} \"\$rom\"" >> /dev/shm/runcommand.log
+
+pushd "$romdir/$system" > /dev/null
+\$jzintv_bin \${options[*]} "\$rom"
+popd
+_EOF_
+    chown $user:$user "$start_script"
+    chmod u+x "$start_script"
 
-    addEmulator 1 "$md_id" "intellivision" "$md_inst/bin/jzintv ${options[*]} %ROM%"
-    options+=(--ecs=1)
-    addEmulator 0 "${md_id}-ecs" "intellivision" "$md_inst/bin/jzintv ${options[*]} %ROM%"
-    addSystem "intellivision"
+    addEmulator 1 "$md_id"     "$system" "bash '$start_script' %XRES% %YRES% %ROM%"
+    addEmulator 0 "$md_id-ecs" "$system" "bash '$start_script' %XRES% %YRES% --ecs=1 %ROM%"
```

Comment:
The file is marked as executable and lives on a native linux fs so you don't need to use "bash" to run it now.

Reply:
Right. Leftover from a sample scriptmodule I peeked on unreflected (dosbox.sh).

Reply:
That's because the script in dosbox is included in `$romdir` which may be on a fat32 fs so this is needed in that case. As the script is moved now it's not needed here.